src/coreclr/jit/typelist.h

```diff
@@ -64,6 +64,7 @@ DEF_TP(SIMD16   ,"simd16" , TYP_SIMD16,  TI_STRUCT,16,16, 16,   4,16, VTF_S|VTF_
 DEF_TP(SIMD32   ,"simd32" , TYP_SIMD32,  TI_STRUCT,32,32, 32,   8,16, VTF_S|VTF_VEC)
 DEF_TP(SIMD64   ,"simd64" , TYP_SIMD64,  TI_STRUCT,64,64, 64,  16,16, VTF_S|VTF_VEC)
 #endif // TARGET_XARCH
+DEF_TP(MASK     ,"mask"   , TYP_MASK,    TI_STRUCT,8, 8,  8,    2,8, VTF_S)
```

Comment:
It is a struct and will eventually map to the `VectorMask<T>` type(s).

Under the hood we have just `k0-k7` which are 64-bits and while the actual hardware manuals don't differentiate further than `k#`, the .NET and C++ intrinsics do. -- .NET has `VectorMask#<T>` while C++ has `__mmask8/16/32/64`. So `VectorMask256<byte>` is equivalent to `__mmask16` since it has 16 entries.

Given that, is just `MASK` sufficient or do we need `MASK8/16/32/64`? If we have more than one, presumably we'd have this be `VTF_S|VTF_MSK`. -- I could see us going either way with there being different tradeoffs for each approach.